pkg/adminapi/client.go

```diff
@@ -61,7 +61,7 @@ func MakeHTTPClient(opts *HTTPClientOpts) (*http.Client, error) {
 		}
 		tlsConfig.RootCAs = certPool
 	}
-	defaultTransport.TLSClientConfig = &tlsConfig
+	defaultTransport.TLSClientConfig = tlsConfig.Clone()
```

Comment:
note: need to take one more look at this race condition, clone the defaultTransport

Reply:
resolved by follow-up: https://github.com/Kong/kubernetes-ingress-controller/issues/1260